rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java

```diff
@@ -95,11 +93,29 @@ public boolean isParentTag() {
     }
 
     private boolean hasGroupAndArtifact(String groupId, @Nullable String artifactId) {
+        return hasGroupId(groupId) && hasArtifactId(artifactId);
+    }
+
+    private boolean hasGroupId(String groupId) {
+        Xml.Tag tag = getCursor().getValue();
+        return groupId.equals(tag.getChildValue("groupId").orElse(model.getGroupId())) ||
+                hasPropertyValue("groupId", groupId);
+    }
+
+    private boolean hasArtifactId(@Nullable String artifactId) {
         Xml.Tag tag = getCursor().getValue();
-        return groupId.equals(tag.getChildValue("groupId").orElse(model.getGroupId())) &&
-                tag.getChildValue("artifactId")
-                        .map(a -> a.equals(artifactId))
-                        .orElse(artifactId == null);
+        return tag.getChildValue("artifactId")
+                .map(a -> a.equals(artifactId))
+                .orElse(artifactId == null) || (artifactId != null && hasPropertyValue("artifactId", artifactId));
+    }
+
+    private boolean hasPropertyValue(String name, String value) {
```

Comment:
This won't capture properties defined in ancestors. We do have recursive property resolution on the `Pom` model already which should probably be used instead.

Reply:
Updated to use model.getValue() - thanks for the suggestion!